.github/workflows/ci.yml

```diff
@@ -0,0 +1,403 @@
+    runs-on: ubuntu-latest
+    outputs:
+      nginx: ${{ steps.generate-nginx-strategy.outputs.versions }}
+    steps:
+      - id: generate-nginx-strategy
+        name: Generate nginx
+        run: |
+          echo "::set-output name=versions::[\"1.17-nginx1-nginx\", \"1.16\", \"1.15\", \"1.14\"]"
+  generate-php-strategy:
+    name: Generate PHP
+    runs-on: ubuntu-latest
+    outputs:
+      php: ${{ steps.generate-php-strategy.outputs.versions }}
+    steps:
+      - id: generate-php-strategy
+        name: Generate PHP
+        run: |
+          echo "::set-output name=versions::[\"7.4\", \"7.3\", \"7.2\"]"
+  generate-php-type-strategy:
+    name: Generate PHP Type
+    runs-on: ubuntu-latest
+    outputs:
+      type: ${{ steps.generate-php-type-strategy.outputs.type }}
+    steps:
+      - id: generate-php-type-strategy
+        name: Generate Type
+        run: |
+          echo "::set-output name=type::[\"cli\", \"fpm\"]"
+  generate-type-strategy:
+    name: Generate Type
+    runs-on: ubuntu-latest
+    needs:
+      - generate-php-type-strategy
+    outputs:
+      type: ${{ steps.generate-type-strategy.outputs.type }}
+    steps:
+      - id: generate-type-strategy
+        name: Generate Type
+        run: |
+          echo "::set-output name=type::[\"cli\", \"fpm\", \"http\"]"
+  lint-docker:
+    name: Lint Dockerfile-${{ matrix.type }}
+    runs-on: ubuntu-latest
+    needs:
+      - generate-type-strategy
+    strategy:
+      fail-fast: false
+      matrix:
+        type: ${{ fromJson(needs.generate-type-strategy.outputs.type) }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Lint Dockerfile-${{ matrix.type }}
+        uses: docker://hadolint/hadolint:latest-debian
+        with:
+          entrypoint: hadolint
+          args: Dockerfile-${{ matrix.type }}
+  lint-shell:
+    name: Lint shell scripts
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - id: files
+        name: Generate shell script file list
+        run: |
+          list=$(ls src/http/nginx/docker* src/php/utils/install-* src/php/utils/docker/* build* test-*  | tr "\n" " " | sed -z '$ s/\n$//')
+          echo -e "::set-output name=list::/github/workspace/${list// / /github/workspace/}build-http.sh"
+      - name: Lint shell scripts
+        uses: docker://koalaman/shellcheck:latest
+        with:
+          args: ${{ steps.files.outputs.list }}
+  build-http:
+    name: Build nginx ${{ matrix.nginx }}
+    needs:
+      - lint-docker
+      - lint-shell
+      - generate-nginx-strategy
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        nginx: ${{ fromJson(needs.generate-nginx-strategy.outputs.nginx) }}
+    steps:
+      - uses: actions/checkout@v2
+      - run: ./build-http.sh $(echo "${{ matrix.nginx }}" | tr '-' ' ')
+      - run: cat ./tmp/image.tags | xargs -I % docker inspect --format='%={{.Id}}:{{index .Config.Env 7}}' %
```

Comment:
this line is failing
<img width="829" alt="image" src="https://user-images.githubusercontent.com/625392/97353198-a1435f80-188b-11eb-9d78-b83d2810195a.png">

Reply:
should be `/tmp/build-....tags`

Reply:
adding `shell: bash` will make this fail

Reply:
@bendavies Adding `shell: bash` to the `build-http.sh` script you mean?

Reply:
Fixed! Thanks for taking a looking into the steps and spotting it @bendavies!

I've also fixed the `build-php` and `build-prometheus-exporter-file` steps, they had similar failures.

Reply:
i suggest adding `shell: bash` to the `run: ...` commands
see https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun and https://github.com/actions/runner/issues/353
adding `shell: bash` will execute:
`bash --noprofile --norc -eo pipefail {0}`, which would have caught the previous error because of `pipefail`

Reply:
@bendavies Thanks for the suggestion, I've added `shell: bash` to the `run` calls, apart from those we call `docker` or `make` commands, as we already lint those somewhere.

Reply:
@bendavies nice catch thanks! Just wondering why the `bash --noprofile --norc -eo pipefail {0}` isn't the default

Reply:
well, the docs say it is the default, but the above linked issue reports that that's not the case.
You can also specify defaults at workflow or job level, instead of specfiying `shell: bash` for each step:
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#defaultsrun
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_iddefaultsrun